Tanvir04Khan/01-max-udemy/03-login-page/src/Components/Login/LoginPage.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useReducer, Fragment } from "react";
+
+import Card from "../UI/Card";
+import UserInput from "./UserInput";
+import Header from "./Header";
+
+import "./LoginPage.css";
+
+const emailReducer = (state, action) => {
+  if (action.type === "email-input") {
+    return {
+      value: action.val,
+      isValid: action.val.includes("@") && action.val.includes(".com"),
+    };
+  }
+  return { value: "", isValid: false };
+};
+
+const passwordReducer = (state, action) => {
+  if (action.type === "password-input") {
+    return { value: action.val, isValid: action.val.trim().length >= 6 };
+  }
+  return { value: "", isValid: false };
+};
+
+const LoginPage = ({ authentication }) => {
+  const [enteredEmail, dispatchEmail] = useReducer(emailReducer, {
+    value: "",
+    isValid: false,
+  });
+  const [enteredPassword, dispatchPassword] = useReducer(passwordReducer, {
+    value: "",
+    isValid: false,
+  });
+
+  const [formIsValid, setFormIsValid] = useState(false);
```

Comment:
boolean variable names should start with:
`is....`
`has......`
`can.....`
`should.....`
etc

Reply:
changed code as per suggestion @faran4engg